game-loop/src/test/java/com/iluwatar/gameloop/BulletTest.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * The MIT License
+ * Copyright © 2014-2019 Ilkka Seppälä
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.iluwatar.gameloop;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Bullet unit test class.
+ */
+public class BulletTest {
+
+  private Bullet bullet;
+
+  @Before
+  public void setup() {
+    bullet = new Bullet();
+  }
+
+  @After
+  public void teardown() {
+    bullet = null;
+  }
+
+  @Test
+  public void testGetPosition() {
+    try {
+      var field = Bullet.class.getDeclaredField("position");
+      field.setAccessible(true);
+      field.set(bullet, 1.5f);
+      Assert.assertEquals(1.5f, bullet.getPosition(), 0);
+    } catch (NoSuchFieldException | IllegalAccessException e) {
+      Assert.fail("Fail to modify field access.");
+    }
+  }
+
+  @Test
+  public void testSetPosition() {
+    try {
+      bullet.setPosition(1.5f);
+      var field = Bullet.class.getDeclaredField("position");
+      field.setAccessible(true);
+      Assert.assertEquals(1.5f, (Float) field.get(bullet), 0);
+    } catch (NoSuchFieldException | IllegalAccessException e) {
+      Assert.fail("Fail to modify field access.");
+    }
+  }
+}
```

Comment:
I don't think these Bullet tests add very much value. It's a simple class with one setter and getter and as such doesn't need to be separately tested. Another thing is the approach to testing where private fields are made accessible. It's not good practise and we should try to find alternate methods.

Reply:
Fixed in commit de763fbc4f138c37212e81e61710c8828e5f43cc